update.sh

```
@@ -34,7 +34,19 @@ IFS=',' read -r -a CONTAINERS <<< "${PLUGIN_CONTAINER}"
 for DEPLOY in ${DEPLOYMENTS[@]}; do
   echo Deploying to $KUBERNETES_SERVER
   for CONTAINER in ${CONTAINERS[@]}; do
+
+    # Collect the last revision number so we can track the next one
+    local LAST=$(kubectl -n ${PLUGIN_NAMESPACE} rollout history deployment/${DEPLOY} |grep -v '$^' |tail -n1 | awk '{print $1}')
+
+    # Initiate the rolloout by modifying the image
     kubectl -n ${PLUGIN_NAMESPACE} set image deployment/${DEPLOY} \
       ${CONTAINER}=${PLUGIN_REPO}:${PLUGIN_TAG} --record
+
+    # Wait for the rollout to complete
+    kubectl -n ${PLUGIN_NAMESPACE} rollout status -r $((++LAST)) --watch
```

How does this behave if you want to update several deployments with different container names ? For example, the following usually prints out errors, but it still updates deployment after deployment since it's trying all container names without exiting.

```
deploy-staging:
    image: quay.io/honestbee/drone-kubernetes
    deployment: [admin, web, resque-scheduler, resque-worker]
    repo: registry.honestbee.com/honestbee/backend
    container: [admin, web, resque-scheduler, resque-worker]
    kubernetes_server: ${KUBERNETES_SERVER_STAGING}
    kubernetes_token: ${KUBERNETES_TOKEN_STAGING}
    tag: ${DRONE_COMMIT_SHA:0:8}
    when:
      branch: [ staging ]
```

---

This is why we're using the local variable LAST to track each one _and_ why were performing the rollouts for each separately and sequentially.

As I noted with the PR, this definitely has its problems.  It doesn't, roll back any previously-applied updates if a later update in the sequence fails.

I figured it would be better to abort the update rather than continue, on error, but if that isn't desired, we can pull out the following `if` check, and it will continue even on error.

---

I'm not entirely sure what would be the best course of action here... Interrupting an upgrade leaves potentially things in an unstable state. Continuing might do the same. @so0k what's your opinion here?

---

I don't think there _is_ a universal best practice here.  In some situations, you would want to abort; in others, you might want to continue.  Personally, I'd rather fail fast... and I'd distrust multipled builds as a matter of course, which is why I defaulted to aborting.  I can definitely see it either way, though.